test/core/files.spec.js

```diff
@@ -49,6 +49,7 @@ describe('files', function () {
       const invalidPath = null
       const stream = ipfs.getReadableStream(invalidPath)
 
+      stream.on('data', () => {})
```

Comment:
Interesting. Something changed for now it requiring to start pumping for the error to be emitted?

Reply:
In the old implementation, converting the pull stream to a readable stream calls [`s.sink(read)`](https://github.com/pull-stream/pull-stream-to-stream/blob/master/index.js#L85) in `pull-stream-to-stream` which calls [`drain()`](https://github.com/pull-stream/pull-stream-to-stream/blob/master/index.js#L97) on the next tick which causes data to start flowing, potentially before the consuming code has set up pipes or data event listeners.

I've opened https://github.com/pull-stream/pull-stream-to-stream/pull/7 as this looks like a bug, at least, it's not consistent with the node docs.

The new implementation doesn't do this conversion so doesn't have the same bug.

Reply:
good catch! 👍